src/renderer/components/input/input.tsx

```diff
@@ -168,32 +175,16 @@ export class Input extends React.Component<InputProps, State> {
           this.validationId = validationId = uniqueId("validation_id_");
         }
         asyncValidators.push(
-          validator.validate(value, this.props).then(
-            () => null, // don't consider any valid result from promise since we interested in errors only
-            error => this.getValidatorError(value, validator) || error,
-          ),
+          validator
+            .validate(value, this.props)
+            .then(() => null)
+            .catch(error => this.getValidatorError(value, validator) || (error instanceof Error ? error.message : String(error))),
         );
       } else {
         if (!validator.validate(value, this.props)) {
           errors.push(this.getValidatorError(value, validator));
         }
       }
-
-      const result = validator.validate(value, this.props);
-
-      if (isBoolean(result) && !result) {
-        errors.push(this.getValidatorError(value, validator));
-      } else if (result instanceof Promise) {
-        if (!validationId) {
-          this.validationId = validationId = uniqueId("validation_id_");
-        }
-        asyncValidators.push(
-          result.then(
-            () => null, // don't consider any valid result from promise since we interested in errors only
-            error => this.getValidatorError(value, validator) || error,
-          ),
-        );
-      }
```

Comment:
Quite heavy code block removed, which is super great. But to make 100% sure we don't need this anywhere?

Reply:
Is caused the duplicate error messages bug. Let me find the issue about that.

Reply:
I cannot find it, but here is a screenshot of the bug (from master)
![Screen Shot 2022-05-17 at 3 47 33 PM](https://user-images.githubusercontent.com/8225332/168897537-ee45aead-845d-4542-a6d3-bf0529c1f7a1.png)